src/logical/model.rs

```diff
@@ -513,26 +531,50 @@ impl Rule {
 
     /// Return the body literals of the rule - immutable.
     #[must_use]
-    pub fn body(&self) -> &Vec<Literal> {
-        &self.body
+    pub fn positive_body(&self) -> &Vec<Atom> {
+        &self.positive_body
+    }
```

Comment:
I've seen several `positive_body().chain(….negative_body())` below – why not add something like this?
```rust
pub fn body(&self) -> impl Iterator<Item = Atom> {
  self.positive_body.iter().chain(self.negative_body.iter())
}
```